packages/serverless/src/model/abstract/document.ts

```diff
@@ -82,20 +83,43 @@ export class ModelDocument extends ModelBasic<DocumentRecord> {
   public async updateOne(
     docId: string,
     document: DocumentRecord,
-    session?: ClientSession
+    session: ClientSession
   ) {
     logger.debug(`ModelDocument::updateDocument()`, { docId });
-    const findDocument = await this.find({ docId });
+    const findDocument = await this.findOne({ docId });
 
-    if (findDocument.length > 0) {
-      const documentRecord: DocumentRecord = {
+    if (findDocument) {
+      const documentRecord = {
         timestamp: new Date(),
         ...document,
-        docId: findDocument[0].docId,
-        deleted: false,
-      } as any;
-
-      return this.collection.insertOne(documentRecord, { session });
+        docId: findDocument.docId,
+        active: true,
+      } as DocumentRecord;
+      // Start transaction
+      session.startTransaction();
```

Comment:
Why do we start session here? We already started transaction, nested transaction is not allowed

Reply:
My bad, I just checked that we have withTx provider cover these commit rollback stuff, I removed and add session into inner method. Solved